plugin/pkg/scheduler/schedulercache/nodeinfocache/node_info_cache_test.go

```diff
@@ -0,0 +1,410 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nodeinfocache
+
+import (
+	"reflect"
+	"testing"
+	"time"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/resource"
+	"k8s.io/kubernetes/plugin/pkg/scheduler/schedulercache"
+)
+
+// TestAssumePod tests that after a pod is assumed, its information is aggregated
+// on node level.
+func TestAssumePod(t *testing.T) {
+	nodeName := "node"
+	tests := []struct {
+		pods []*api.Pod
+
+		wNodeInfo *schedulercache.NodeInfo
+	}{{
+		pods: []*api.Pod{makeBasePod(nodeName, "test", "100m", "500", []api.ContainerPort{{HostPort: 80}})},
+		wNodeInfo: &schedulercache.NodeInfo{
+			RequestedResource: &schedulercache.Resource{
+				MilliCPU: 100,
+				Memory:   500,
+			},
+			PodNum: 1,
+		},
+	}, {
+		pods: []*api.Pod{
+			makeBasePod(nodeName, "test-1", "100m", "500", []api.ContainerPort{{HostPort: 80}}),
+			makeBasePod(nodeName, "test-2", "200m", "1Ki", []api.ContainerPort{{HostPort: 8080}})},
+		wNodeInfo: &schedulercache.NodeInfo{
+			RequestedResource: &schedulercache.Resource{
+				MilliCPU: 300,
+				Memory:   1524,
+			},
+			PodNum: 2,
+		},
+	}}
+
+	for i, tt := range tests {
+		cache := newNodeInfoCache(time.Second, time.Second, nil)
+		for _, pod := range tt.pods {
+			err := cache.AssumePod(pod)
+			if err != nil {
+				t.Fatalf("AssumePod failed: %v", err)
+			}
+		}
+		n := cache.getNodeInfo(nodeName)
+		if !reflect.DeepEqual(n, tt.wNodeInfo) {
+			t.Errorf("#%d: node info get=%s, want=%s", i, n, tt.wNodeInfo)
+		}
+	}
+}
+
+// TestExpirePod tests that assumed pods will be removed if expired.
+// The removal will be reflected in node info.
+func TestExpirePod(t *testing.T) {
+	nodeName := "node"
+	now := time.Now()
+	ttl := 10 * time.Second
```

Comment:
The idea occurred to me.

Let's walk through the current architecture. There are two stores: `ScheduledPodLister` and `assumedPods`. If we can combine these two stores, then it's exactly what I am proposing here -- the scheduler cache.

@lavalamp also mentioned above that we could replace the system modeler instead of having parallel structs.

However, such reconstruction needs a lot more work. This PR is the minimal change that I need to make #18831 work.

I'm pretty open to the scope of the problem and do more work. I'm just not sure what everyone's goal is. I was going to talk to @lavalamp this week since he just came back to bay area.